python/lsst/analysis/ap/plotImageSubtractionCutouts.py

```diff
@@ -50,6 +49,44 @@
 from . import apdb
 
 
+class ButlerCache:
+    """Global class to handle butler queries, to allow lru_cache and
+    multiprocessing.pool to work together.
+
+    If we redo this all to work with BPS or other parallelized systems, or get
+    good butler-side caching, we could remove this lru_cache system.
+    """
+    def set(self, butler, config):
+        """Call this to store a Butler and pex.Config instance before using the
+        global class instance.
+        """
+        self._butler = butler
+        self._config = config
+        # Ensure the caches are empty if we've been re-set.
+        self.get_exposures.cache_clear()
+        self.get_catalog.cache_clear()
+
+    @functools.lru_cache(maxsize=4)
+    def get_exposures(self, instrument, detector, visit):
+        """Return science, template, difference exposures, using a small
+        cache so we don't have to re-read files as often.
+        """
+        dataId = {'instrument': instrument, 'detector': detector, 'visit': visit}
+        return (self._butler.get(self._config.science_image_type, dataId),
+                self._butler.get(f'{self._config.diff_image_type}_templateExp', dataId),
+                self._butler.get(f'{self._config.diff_image_type}_differenceExp', dataId))
+
+    @functools.lru_cache(maxsize=4)
+    def get_catalog(self, instrument, detector, visit):
+        """Return the diaSrc catalog from the butler."""
```

Comment:
and here. (otherwise maybe explicitly mark this class as private)

Reply:
Have also made it private but prepending a `_`.